User menu

claudiodekker opened the pull request:
This should (hopefully) mark #45 as completed.

nekodex commented:
Found an issue - if you repeatedly click on your avatar to show/hide the dialog too fast, the page progressively gets dimmer and dimmer.

claudiodekker commented:
Well spotted. Turns out that's an issue already in existence long before this feature was added, namely with the login modal. (Look at the currently live site, which doesn't include this commit)

To be fair, I don't think this issue is *that* important, as it doesn't block user-experience, but I'll still take a look at it, as it should also resolve the issue on the guest-side of things.

If possible, I'll see if I can create a separate bugtracker-issue for this & claim it, although I'm not sure if I've got the rights to do so.

nanaya commented:
- ctb (and other play mode) logo is available as font `<i class="fa osu fa-ctb-o"></i>`
- country flags are in `images/flags/`
- check this regarding css: #53
- apparently having the modal button in front of overlay breaks the overlay ;_;
- try using localization when possible

claudiodekker commented:
@nanaya Alright, thanks. I'll get onto it in a bit. 

Is there any quick(er) way for me to reach you regarding quick reviews/feedback on things, or should I just be a patient man? ;)

nanaya commented:
you can reach me in slack. How fast I will response is a bit of mystery though.

claudiodekker commented:
Fixed most issues mentioned, and did some (on slack) discussed stuff..

- Replaced gamemode icon (png) with the font version.
- Now including country flags automatically (NOTE: I'm using the path "/images/flags/{{ Auth::user()->country_acronym }}.png", let me know if this is correct or not)
- Reworked the ENTIRE dropdown modal for both login and when logged in, to be based upon the BEM concept.
- Now using localization

As discussed with @nanaya on Slack, this entire has been reworked to be based upon the BEM concept:
This also means that there is now a BEM folder, and a bem-index.less file, in which each BEM-block is being included by it's own .less file.
This can be changed in the future, but for now I suppose this is okay. It's pretty much a playtest for now, doesn't really matter as it's easily moveable.

nanaya commented:
sorry for late response,

- there are merge conflicts
- files should have end of lines
- less files will have two spaces indentations from now on (following common style)

nanaya commented on resources/assets/less/bem/badges.less:
```diff
@@ -0,0 +1,48 @@
+		text-align: center;
+		display: block;
+
+		&--small {
```
just fold into `badges__badge`?

claudiodekker replied:
The reason I did this, is to create a R-E-M structure, nested as such.

nanaya commented on resources/assets/less/bem/badges.less:
```diff
@@ -0,0 +1,48 @@
+		display: block;
+
+		&--small {
+			background-size: 38px;
```
`contain` will work here

nanaya commented on resources/assets/less/bem/badges.less:
```diff
@@ -0,0 +1,48 @@
+		}
+
+		&--level {
+			.at2x("/images/badges/levelbadge.png");
```
`.at2x` will override background-size at retina.

claudiodekker replied:
Ah, I see. Right. Guess we'll have to '!important' on the --small one then.

nanaya commented on resources/assets/less/bem/bootstrap/modal-body.less:
```diff
@@ -0,0 +1,79 @@
+*
+*/
+
+.modal-body {
```
probably better use a different class name since it's not obvious it's overriding bootstrap styling.

claudiodekker replied:
Actually, it is. Notice the file's path name.
Besides, normal REM-blocks use a different structure (.modal__body)

nanaya commented on resources/assets/less/bem/bootstrap/modal-body.less:
```diff
@@ -0,0 +1,79 @@
+	// Elements
+	&__compartment {
+		display: flex;
+		height: 140px;
```
I don't think fixed height should be used

claudiodekker replied:
Absolutely right.

nanaya commented on resources/assets/less/bem/bootstrap/modal-body.less:
```diff
@@ -0,0 +1,79 @@
+	&__compartment {
+		display: flex;
+		height: 140px;
+		padding: 10px;
```
`@grid-gutter-width/4`?

nanaya commented on resources/assets/less/bem/bootstrap/modal-header.less:
```diff
@@ -0,0 +1,81 @@
+	}
+
+	&__dimmer {
+		background: -moz-linear-gradient(top, rgba(0,0,0,0.5) 0%, rgba(0,0,0,0.75) 100%); /* FF3.6+ */
```
no need for old browsers compatibility

claudiodekker replied:
No kittens were killed while generating this code.

nanaya replied:
my eyes kind of harmed +_+ also changing it is going to be difficult.

And please use `fade(#abc, x%)` instead of rgba.

nanaya commented on resources/assets/less/bem/bootstrap/modal-header.less:
```diff
@@ -0,0 +1,81 @@
+	}
+
+	&__userinfo {
+		height: 49px;
```
why fixed height

nanaya commented on resources/assets/less/bem/bootstrap/modal-header.less:
```diff
@@ -0,0 +1,81 @@
+		align-items: flex-end;
+		.at2x("/images/headers/profile-covers/c3.jpg", 430px, 70px);
+		background-size: 430px auto;
+		background-position: center center;
```
apparently single value means same value for x and y

claudiodekker replied:
Is that in the spec / does it work cross browser?

nanaya replied:
apparently [the actual behavior](https://developer.mozilla.org/en-US/docs/Web/CSS/position_value) is the other value is always 50% (x or y if keyword, always y if value)

nanaya commented on resources/assets/less/bem/bootstrap/modal.less:
```diff
@@ -0,0 +1,50 @@
+}
+
+// Javascript-based hooks/styling
+.js-user-dropdown-modal { 
```
this probably shouldn't be here

claudiodekker replied:
Hm, you might be right, however I can't seem to think of a more suitable place.

It's related to the modal as it [already was] (https://github.com/ppy/osu-web/blob/master/resources/assets/less/login-modal.less), so I think this should be okay-ish?

nanaya replied:
I think we should expect everything here is from same block otherwise it'll get fun real fast.

claudiodekker replied:
Yeah, probably, although since we're already having a custom prefix for `js-`, which might be styled sometimes (like now), then the only other solution is creating new files, which I don't know is a good idea when you (maybe in the future) port the entire CSS to BEM-style.

Until this is decided, I think it's safe to simply keep it like this..

nanaya replied:
I think we'll keep at least few files outside of bem anyway (utilities, variables, globals).

claudiodekker replied:
Very well, moving it.

nanaya commented on resources/assets/less/bem/quick-info.less:
```diff
@@ -0,0 +1,60 @@
+		padding: 0;
+		width: 100%;
+
+		> li {
```
`> *` probably better here (no increase in specificity)

nanaya commented on resources/assets/less/bem/quick-info.less:
```diff
@@ -0,0 +1,60 @@
+
+			.text-right {
+				font-weight: 600;
+				text-align: right;
```
pretty sure not needed

claudiodekker replied:
lol, correct. #bootstrap

nanaya commented on resources/assets/less/bem/quick-info.less:
```diff
@@ -0,0 +1,60 @@
+			display: flex;
+			justify-content: space-between;
+
+			.text-right {
```
`&__statistics--value`?

claudiodekker replied:
Yeah, the idea behind this, is that you can go overboard on custom BEM files. I think this really should be discussed.

The problem here is, is that I could create another item on the element `quick-info-statistics__statistic` called something like `quick-info-statistics-statistic`, with the `_statistics--value` can be `quick-info-statistics-statistic--value`.

See what I'm getting at? This happens at one or two more places (maybe).

nanaya replied:
oh right, should be just `user-statistics__value`. Reusable in profile page as well.

claudiodekker replied:
So, `<li class="quick-info-statistics__statistic user-statistics"><span>{{ trans("users.show.stats.ranked_score") }}</span><span class="user-statistics__value">{{ number_format(29955840356) }}</span></li>` then?

nanaya replied:
there isn't anything in `quick-info-statistics__statistic`, is there?

claudiodekker replied:
`quick-info-statistics` is the block/wrapper element. `quick-info-statistics__statistics` is the bit at line 37.
But no, there isn't anything in `user-statistics`.

nanaya replied:
display and justify content probably belongs to user statistics.

claudiodekker replied:
Actually, this is being used fo the following: 
![screenshot 2015-09-14 04 51 02](https://cloud.githubusercontent.com/assets/1752195/9852884/df8d2c24-5b00-11e5-9ba2-b936327a2b44.png)

nanaya commented on resources/assets/less/bem/quick-info.less:
```diff
@@ -0,0 +1,60 @@
+	&__level {
+		font-style: normal;
+		color: #555;
+		font-family: @font-family-sans-serif;
```
?

claudiodekker replied:
¿

That's used to pick the Exo v2 w/ fallback or whatever the font is, instead of typing it out fully.

nanaya replied:
isn't it the default already?

claudiodekker replied:
It wasn't for this H1, so it's actually overriding something. But I suppose this'll be changed considering *that* will be changed.

nanaya commented on resources/views/objects/user-dropdown.blade.php:
```diff
@@ -16,50 +16,90 @@
+					</div>
+					<div class="modal-body modal-body--compartimentalized">
+						<div class="modal-body__compartment modal-body__compartment--left quick-info">
+							<h1 class="quick-info__level">Level 5</h1>
```
this isn't a heading.

claudiodekker replied:
Right. Line 51 is? (`<h1 class="userinfo-small__username">{{ Auth::user()->username }}</h1>`)

nanaya commented on resources/assets/less/bem/bootstrap/modal-body.less:
```diff
@@ -0,0 +1,79 @@
+*/
+
+.modal-body {
+	background-color: #FFF;
```
this overrides styling used in forum "search" modal

claudiodekker replied:
Ah, right, didn't look at this one properly..

nanaya commented on resources/lang/en/users.php:
```diff
@@ -63,7 +60,8 @@
         ],
         'first_members' => 'here since the beginning',
-        'is_supporter' => 'osu!supporter',
+        'is_supporter' => 'osu!Supporter',
```
oh right, this and one blow should be in all lowercase. The one in design image is wrong.
